src/metatags/opportunityHandler.js

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { TITLE, DESCRIPTION, H1 } from './constants.js';
+
+function removeTrailingSlash(url) {
+  return url.endsWith('/') ? url.slice(0, -1) : url;
+}
+
+/**
+ * Synchronizes existing suggestions with new data
+ * by removing outdated suggestions and adding new ones.
+ *
+ * @param {Object} params - The parameters for the sync operation.
+ * @param {Object} params.opportunity - The opportunity object to synchronize suggestions for.
+ * @param {Array} params.newData - Array of new data objects to sync.
+ * @param {Function} params.buildKey - Function to generate a unique key for each item.
+ * @param {Function} params.mapNewSuggestion - Function to map new data to suggestion objects.
+ * @param {Object} params.log - Logger object for error reporting.
+ * @returns {Promise<void>} - Resolves when the synchronization is complete.
+ */
+export async function syncSuggestions({
```

Comment:
I wrote already a general `syncSuggestions` function as a util function here: https://github.com/adobe/spacecat-audit-worker/blob/main/src/utils/data-access.js#L51

maybe it is possible for you to re-use it? Also feel free to make improvements to it. But since all audits have kind of the same logic for this, it might make sense to have one common function.

Reply:
Thank you for pointing that out. I did initially refer to your syncSuggestions function and used parts of it. 
However, there are some customisations done in my method specifically for meta-tags that may not be possible to do in your method while ensuring compatibility with the ongoing work others are doing for audit opportunities, and I decided it would be safer to write a separate function. I’ve also renamed it to avoid any confusion.

Reply:
As discussed with @dzehnder, we can do this refactoring as a follow-up, once @dipratap is back from vacation next week.

We can merge without this refactoring to unblock the UI work